spring-cloud-stream-schema/src/main/java/org/springframework/cloud/stream/schema/avro/RecordNamingStrategy.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.schema.avro;
+
+import org.apache.avro.Schema;
+
+/**
+ * @author José A. Íñigo
+ */
+public class RecordNamingStrategy implements SubjectNamingStrategy {
```

Comment:
`RecordNamingStrategy` sounds very generic. Since this is a custom implementation, we probably want to be more descriptive in the name. Interestingly, a class with the same implementation is used in [tests only](https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-schema/src/test/java/org/springframework/cloud/schema/avro/CustomSubjectNamingStrategy.java). Could you remove that test class and its usage and use the new one instead? We should add a note about this implementation in the docs as well. I think this implementation sounds like a common use case and therefore it's okay to go into the core framework.  However, if more such implementations are needed, the expectation is that the applications create their own implementations and re-use them by overriding the property.